code/items/equipments/entities/C4Entity.cs

```diff
@@ -53,45 +104,111 @@ public override void Simulate(Client cl)
         {
             if (IsClient)
             {
-
                 if (CreatedDisplay)
                 {
                     //No way to parent a world panel to an entity :(
                     //We need to find a better way of doing this.
                     //I heard supposedly you can use FrameSimulate, but I was getting weird duplicated results (think Windows XP window glitch)
-                    BombDisplay.Transform = GetAttachment("timer") ?? Transform;
-                    BombDisplay.WorldScale = 0.1f;
+                    TimerDisplay.Transform = GetAttachment("timer") ?? Transform;
+                    TimerDisplay.WorldScale = 0.05f;
 
                     return;
                 }
                 else
                 {
-                    BombDisplay = new WorldPanel();
+                    TimerDisplay = new WorldPanel();
                     CreatedDisplay = true;
 
-                    BombDisplay.AddClass("c4display");
-                    BombDisplay.StyleSheet.Add(StyleSheet.FromFile("/ui/alivehud/c4/c4Display.scss"));
+                    TimerDisplay.AddClass("c4worldtimer");
+                    TimerDisplay.StyleSheet.Add(StyleSheet.FromFile("/ui/alivehud/c4/C4WorldTimer.scss"));
+
+                    TimerDisplayLabel = TimerDisplay.AddChild<Label>();
+                    TimerDisplayLabel.Text = "--:--";
```

Comment:
Couldn't that be made as it's own `WorldPanel` class?

Reply:
Here, you aren't returning as well

Reply:
> Couldn't that be made as it's own `WorldPanel` class?

If you're referring to implementing the bomb display stuff in a class inheriting by WorldPanel, not really necessary. Easier just to create it by code.